design/token.go

```diff
@@ -13,18 +13,36 @@ var externalToken = a.MediaType("application/vnd.externalToken+json", func() {
 		a.Attribute("access_token", d.String, "The token associated with the identity for the specific external provider")
 		a.Attribute("scope", d.String, "The scope associated with the token")
 		a.Attribute("token_type", d.String, "The type of the toke, example : bearer")
-		a.Attribute("username", d.String, "The username of the identity loaded from the specific external provider. Optional attribute.")
-		a.Required("access_token", "scope", "token_type")
+		a.Attribute("username", d.String, "The username of the identity loaded from the specific external provider")
+		a.Attribute("provider_url", d.String, "The external provider URL.")
```

Comment:
Should we call this this `api_url` or maybe `provider_api_url` just to be consistent because it is really the API url we are turning.

Reply:
Not sure actually but OK - https://github.com/fabric8-services/fabric8-auth/pull/335/commits/fbf9b742b9b1f8a8500cfdd8416e6fa7669d5641